packages/providers/src/test-contract/src/main.sw

```diff
@@ -8,29 +8,29 @@ struct TestStruct {
 }
 
 abi TestContract {
-  fn foo(gas_: u64, amount_: u64, coin_: b256, value: u64) -> u64;
-  fn boo(gas_: u64, amount_: u64, coin_: b256, value: TestStruct) -> TestStruct;
-  fn barfoo(gas_: u64, amount_: u64, coin_: b256, value: u64) -> u64;
-  fn foobar(gas_: u64, amount_: u64, coin_: b256, value: ()) -> u64;
+  fn foo(value: u64) -> u64;
+  fn boo(value: TestStruct) -> TestStruct;
+  fn barfoo(value: u64) -> u64;
+  fn foobar(value: ()) -> u64;
 }
 
 impl TestContract for Contract {
-  fn foo(gas_: u64, amount_: u64, asset_id_: b256, value: u64) -> u64 {
+  fn foo(value: u64) -> u64 {
     log_u64(3735928559);
     value
   }
-  fn boo(gas_: u64, amount_: u64, asset_id_: b256, value: TestStruct) -> TestStruct {
+  fn boo(value: TestStruct) -> TestStruct {
     log_u64(3735928559);
     log_u64(value.b);
     TestStruct {
       a: !value.a,
       b: value.b + 1,
     }
   }
-  fn barfoo(gas_: u64, amount_: u64, asset_id_: b256, value: u64) -> u64 {
+  fn barfoo(value: u64) -> u64 {
     63
   }
-  fn foobar(gas_: u64, amount_: u64, asset_id_: b256, value: ()) -> u64 {
+  fn foobar(value: ()) -> u64 {
```

Comment:
There is no reason to keep the unit type `value: ()` anymore. It can be dropped.

Reply:
We should still have a test for that though since it _is_ a valid parameter type.

Reply:
It's valid to keep this, for validate empty params?

Reply:
Yes that makes sense. Maybe we should keep one version with no parameters and one with an empty parameter.